extension/storage/filestorage/config.go

```diff
@@ -20,6 +20,10 @@ type Config struct {
 
 	// FSync specifies that fsync should be called after each database write
 	FSync bool `mapstructure:"fsync,omitempty"`
+
+	// CreateDirectory specifies that the directory should be created automatically by the extension on start
+	CreateDirectory bool   `mapstructure:"create_directory,omitempty"`
+	DirectoryPerm   string `mapstructure:"directory_permission,omitempty"`
```

Comment:
Why don't we spell out the full word, to prevent users from wondering if it's `Permissions`, `Permanent` or `Permeable`. :upside_down_face: 

Also is it just me, or does `directory_permissions` sound better than `directory_permission`? It is more than one permission after all. 

```suggestion
	DirectoryPermissions   string `mapstructure:"directory_permissions,omitempty"`
```